### DISCUSSION TITLE
Demo question template

### DISCUSSION THREAD

**Discussion by mrdbourke** [mrdbourke/pytorch-deep-learning#117] [General]
This template is to help with the formatting of questions to best help others who are reading it.

The four main sections are:
1. **Where you're stuck** - best to put a timestamp/video number of the course here so people know exactly where you are (e.g. ZTM video 135, timestamp 11:07)
2. **Your problem** - what is your problem?
3. **Your code** - what code did you try that isn't working?
4. **What you've tried so far** - detailing the troubleshooting steps you've taken already will help others help you.

You can copy the demo template below and fill out the sections in your question, much of the formatting has been done just to make things look nicer:

```
* Course video/timestamp: (e.g. video number 147 on ZTM, timestamp 10:35)

## My error

TODO: Add some info here about your error...

## My code

TODO: Add your code here, best to format with backticks as well, for example: 

\```python <- use triple backticks before and after your code, write "python" after the first set to make the code formatted
[your code here]
\``` <- be sure to delete the slashes, only keep the backticks

## What I've tried so far

TODO: Add some steps for what you've tried to do so far to solve your error... (this will help others know what you've done when troubleshooting) 
```

**Comment by chos1npc** [comment]
* Course video/timestamp: (chapter01.load model)
* 
## My code

I try to turn
# To load in a saved state_dict we have to instantiate a new instance of our model class
loaded_model_0 = LinearRegressionModel()

# Load the saved state_dict of model_0 (this will update the new instance with updated parameters)
loaded_model_0.load_state_dict(torch.load(f=MODEL_SAVE_PATH))
to
loaded_model_0 = torch.load(f=MODEL_SAVE_PATH)
loaded_model_0
because I think if I use model in other program, I don't have it's a "LinearRegressionModel",so I use this code to load model, I don't know what happen for that error can't do 'state_dict' and 'eval'
THX

**Comment by chos1npc** [comment]
* Course video/timestamp: (e.g. video number 94 on youtube)

## My error

TODO: Add some info here about your error...

## My code

TODO: Add your code here, best to format with backticks as well, for example: 
class FashionMNISTModelV1(nn.Module):
  def __init__(self,
               input_shape: int,
               hidden_units: int,
               output_shape: int):
    super().__init__()
    self.layer_stack = nn.Sequential(
        nn.Flatten(), # flatten inputs into a single vector
        nn.Linear(in_features=input_shape,
                  out_features=hidden_units),
        nn.ReLU(),
        nn.Linear(in_features=hidden_units,
                  out_features=output_shape),
        nn.ReLU()
    )
  
  def forward(self, x: torch.Tensor):
    return self.layer_stack(x)
def train_step(model: torch.nn.Module,
               data_loader: torch.utils.data.DataLoader,
               loss_fn: torch.nn.Module,
               optimizer: torch.optim.Optimizer,
               accuracy_fn,
               device: torch.device = device):
  """Performs a training with model trying to learn on data_loader."""
  train_loss, train_acc = 0, 0

  # Put model into training mode
  model.train()

  # Add a loop to loop through the training batches
  for batch, (X, y) in enumerate(data_loader):
    # Put data on target device 
    X, y = X.to(device), y.to(device)

    # 1. Forward pass (outputs the raw logits from the model)
    y_pred = model(X)
    
    # 2. Calculate loss and accuracy (per batch)
    loss = loss_fn(y_pred, y)
    train_loss += loss # accumulate train loss
    train_acc += accuracy_fn(y_true=y,
                             y_pred=y_pred.argmax(dim=1)) # go from logits -> prediction labels
    
    # 3. Optimizer zero grad
    optimizer.zero_grad()
    
    # 4. Loss backward
    loss.backward()
    
    # 5. Optimizer step (update the model's parameters once *per batch*)
    optimizer.step()
  
  # Divide total train loss and acc by length of train dataloader
  train_loss /= len(data_loader)
  train_acc /= len(data_loader)
  print(f"Train loss: {train_loss:.5f} | Train acc: {train_acc:.2f}%")


\```python <- use triple backticks before and after your code, write "python" after the first set to make the code formatted
[your code here]
\``` <- be sure to delete the slashes, only keep the backticks

## What I've tried so far
why can use the ReLU in output, what it's funtion?
I try to use softmax in output , but get worst result

**Comment by Landry-w** [comment]
Hi Daniel, I'm having problems with pytorch versions because I'm using the torch-2.1.1 torchvision-0.16.1 packages and the codes in the tutorial don't really work. Would you like to help me fix it?

**Comment by MubasharaRehman** [comment]
I followed this video, code is exactly same but when i am trying this chunk of code, 
........................................................................................................................
# torch.manual_seed(42)
# An Epoch is one loop through the data...
epochs = 1

# TRAINING LOOP
# Step 0 >> Loop through the data
for epoch in range(epochs): 
  model_0.train()
  # Step 1 >> fORWARD PASS
  y_pred = model_0(X_train)
  # Step 2 >> CALCULATE THE LOSS
  loss = loss_fn( y_pred, y_train )
  # Step 3 >> OPTIMIZER ZERO GRAD
  optimizer.zero_grad()
  # Step 4 >> PERFORM BACKPROPAGATION ON THE LOSS WITH RESPECT TO THE PARAMETERS OF THE MODEL
  loss.backward()
  # Step 5 >> STEP THE OPTIMIZER (PERFORMS GRADIENT DESCENT)
  optimizer.step()
model_0.eval()  # turns off gradient tracking

/////////////////////////////////////////////////////////////////

i get error on this line,
---> 12   loss = loss_fn( y_pred, y_train )

RuntimeError: Boolean value of Tensor with more than one value is ambiguous

I am new learner, kindly help me where is the mistake and how to fix it
